juju_verify/exceptions.py

```diff
@@ -24,3 +29,17 @@ class VerificationError(Exception):
 
 class CharmException(Exception):
     """Exception related to the charm or the unit that runs it."""
+
+
+class ActionFailed(Exception):
+    """Exception related to failing action run on the unit."""
+
+    def __init__(self, error: JujuError, unit: Unit, action: str,
+                 params: Optional[Dict[str, Any]] = None):
+        """Initialize ActionFailed error message."""
+        params = params or {}
+        params_str = " ".join(f"{name}={value}" for name, value in params.items())
+        juju_error_message = os.linesep.join(f"  {err}" for err in error.errors)
+        self.message = f"{unit.entity_id}: action `{action}{params_str}` failed " \
```

Comment:
Is it intended that `{action}` and `{params_str}` don't have a space delimiter between them?

Reply:
No that is wrong. I will fix it.